pyacq/core/nodegroup.py

```diff
@@ -56,38 +31,57 @@ class _NodeGroup(RPCServer):
     """
     def __init__(self, name, addr):
         RPCServer.__init__(self, name, addr)
+        self.app = QtGui.QApplication([])
         self.nodes = {}
+
+    def run_forever(self):
+        self._readsocket_thread = ThreadReadSocket(self, parent = None)
+        self._readsocket_thread.new_message.connect(self._mainthread_process_one)
+        self._readsocket_thread.start()
+        self.app.exec_()
+    
+    def _mainthread_process_one(self, name, msg):
+        self._readsocket_thread.wait()
+        self._process_one(bytes(name), bytes(msg))
+        if self.running():
+            self._readsocket_thread.start()
```

Comment:
I imagine there is some overhead for starting / stopping the thread for every message. 
Maybe better to have the thread run in a loop and send it a message from the main thread when it needs to stop.